utils/evaluate_utils.py

```diff
@@ -99,6 +99,149 @@ def class_PR_curves(
+                    single_cell_data["Mitocheck_Phenotypic_Class"] != phenotypic_class
+                ]
+            )
+            .sample(positive_label_indexes.shape[0], random_state=0)
+            .index
+        )
+        # the new class training data are the two subsets found above
+        # this new class training data will have equal numbers of positive and negative labels
+        # this removes the drastic class imbalances
+        single_cell_data = single_cell_data.loc[
+            positive_label_indexes.union(negative_label_indexes)
+        ]
+
+    model_classes = single_class_model.classes_
+    X, y = get_X_y_data(single_cell_data, feature_type)
+
+    # predict class probabilities for feature data
+    y_probas = single_class_model.predict_proba(X)
+
+    # figure out where to plot PR curves
+    # ax_x is x axis coordinate for positive label PR curve, ax_y is y axis coordinate for positive label PR curve
+    # negative label PR curves go below positive label (so ax_x+1)
+    ax_x = (
+        int(phenotypic_class_index / 5) * 2
+    )  # multiply by 2 because every positive label has a negative label below it
+    ax_y = phenotypic_class_index % 5  # modulus gives us y coordinate
+
+    color_key = f"{feature_type} ({evaluation_type})"
+    plot_color = data_split_colors[color_key]
+
+    for index, model_class in enumerate(model_classes):
+        # get precision, recall, threshold values for the model class (positive or negative label)
+        precision, recall, thresholds = precision_recall_curve(
+            y, y_probas[:, index], pos_label=model_class
+        )
+        # last values in precision/recall curve don't correspond to cell dataset
+        thresholds = np.append(thresholds, None)
+
+        # plot PR data
+        axs[ax_x + index, ax_y].plot(
+            recall, precision, label=color_key, color=plot_color
+        )
+        # set title and axis labels for subplot
+        axs[ax_x + index, ax_y].set_title(model_class)
+        axs[ax_x + index, ax_y].set(xlabel="Recall", ylabel="Precision")
+
+        # add pr data to compiled dataframe
+        PR_data.append(
+            pd.DataFrame(
+                {
+                    "Model_Class": model_class,
+                    "PR_Threshold": thresholds,
+                    "Precision": precision,
+                    "Recall": recall,
+                    "data_split": evaluation_type,
+                    "shuffled": "shuffled" in model_type,
+                    "feature_type": feature_type,
+                }
+            )
+        )
+
+    # only label outer plots
+    for ax in axs.flat:
+        ax.label_outer()
+
+    # add legend to figure with all subplots
+    handles, labels = ax.get_legend_handles_labels()
+    fig.legend(handles, labels, loc="upper right")
+
+    # compile PR data
+    # some thresholds are None because last PR value doesnt correspond to cell dataset (these values are always P=1, R=0), remove these rows from PR data
```

Comment:
```suggestion
    # some thresholds are None because last PR value doesn't correspond to cell dataset (these values are always P=1, R=0), remove these rows from PR data
```

Reply:
Added to notebook in https://github.com/WayScience/phenotypic_profiling_model/commit/987e1ec5a50e2751ae962ec53d06265164fcb301